.github/workflows/publish.yml

```diff
@@ -6,9 +6,20 @@ on:
     tags: ['*']
 
 jobs:
+  version-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Check module version
+        # We need this step to run only on push with tag.
+        if: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags/') }}
```

Comment:
And then we can use `if` condition on job-level here

Reply:
Moved to job-level